ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/validator/PostSyncCommitteeData.java

```diff
@@ -0,0 +1,128 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.ethereum.json.types.validator;
+
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.INTEGER_TYPE;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.UINT64_TYPE;
+
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import it.unimi.dsi.fastutil.ints.IntSet;
+import java.util.List;
+import java.util.Objects;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+
+public class PostSyncCommitteeData {
+  private int validatorIndex;
+  private IntSet syncCommitteeIndices;
+  private UInt64 untilEpoch;
+
+  public static final DeserializableTypeDefinition<PostSyncCommitteeData>
+      SYNC_COMMITTEE_SUBSCRIPTION =
+          DeserializableTypeDefinition.object(PostSyncCommitteeData.class)
+              .name("PostSyncCommitteeData")
+              .initializer(PostSyncCommitteeData::new)
+              .withField(
+                  "validator_index",
+                  INTEGER_TYPE,
+                  PostSyncCommitteeData::getValidatorIndex,
+                  PostSyncCommitteeData::setValidatorIndex)
+              .withField(
+                  "sync_committee_indices",
+                  DeserializableTypeDefinition.listOf(INTEGER_TYPE),
+                  PostSyncCommitteeData::getSyncCommitteeIndices,
+                  PostSyncCommitteeData::setSyncCommitteeIndices)
+              .withField(
+                  "until_epoch",
+                  UINT64_TYPE,
+                  PostSyncCommitteeData::getUntilEpoch,
+                  PostSyncCommitteeData::setUntilEpoch)
+              .build();
+
+  public PostSyncCommitteeData() {}
+
+  public PostSyncCommitteeData(
+      final int validatorIndex, final IntSet syncCommitteeIndices, final UInt64 untilEpoch) {
+    this.validatorIndex = validatorIndex;
+    this.syncCommitteeIndices = syncCommitteeIndices;
+    this.untilEpoch = untilEpoch;
+  }
+
+  public PostSyncCommitteeData(
+      final SyncCommitteeSubnetSubscription syncCommitteeSubnetSubscription) {
+    this.validatorIndex = syncCommitteeSubnetSubscription.getValidatorIndex();
+    this.syncCommitteeIndices = syncCommitteeSubnetSubscription.getSyncCommitteeIndices();
+    this.untilEpoch = syncCommitteeSubnetSubscription.getUntilEpoch();
+  }
+
+  public SyncCommitteeSubnetSubscription toSyncCommitteeSubnetSubscription() {
```

Comment:
`SyncCommitteeSubnetSubscription` could become a `record` but happy to have this done separately

Reply:
true, we haven't used record anywhere so i didnt action that suggestion when intellij told me...